File: leetcode/top-interview-150/88/Java/Solution.java
public class Solution {

    public static void merge(int[] nums1, int m, int[] nums2, int n) {
        int k = m + n - 1; // length of the final array
        int i = m - 1;
        int j = n - 1;

        while ( j >= 0){
            if (i > 0 && nums1[i] > nums2[j]){
                nums1[k] = nums1[i];
                i--;
            } else {
                nums1[k] = nums2[j];
                j--;
            }
            k--;
        }
    }

    public static void main(String[] args) {
        int[] nums1 = {1,2,3,0,0,0};
        int m = 3;
        int[] nums2 = {2,5,6};
        int n = 3;
        merge(nums1,m,nums2,n);
        for (int i : nums1) {
            System.out.println(nums1[i]);
        }
    }
    
}

File: leetcode/top-interview-150/88/Solution.go
package main

import "fmt"

func merge(nums1 []int, m int, nums2 []int, n int) {
	k := m + n - 1 // last index of nums1 array
	i := m - 1
	j := n - 1

	for j >= 0 {
		if i >= 0 && nums1[i] >= nums2[j] {
			nums1[k] = nums1[i]
			i--
		} else {
			nums1[k] = nums2[j]
			j--
		}
		k--
	}

	for _, num := range nums1 {
		fmt.Println(num)
	}
}

func main() {
	nums1 := []int{2, 0}
	nums2 := []int{1}
	m := 1
	n := 1
	merge(nums1, m, nums2, n)
}
